### PR TITLE
feat(node): add e2e project generator

### PR DESCRIPTION
**Pull request by jaysoo**
This PR adds a `@nrwl/node:e2e-project` generator to configure a server or CLI test suite for Node apps.

Also adds `--e2eTestRunner` for `@nrwl/node:app` generator, which generates tests by default.

Users can generate a new app `nx g @nrwl/node:app api --framework=fastify` and then run:

```bash
nx serve api
nx e2e api-e2e
```

Both the app and e2e tests use `PORT` environment variable, so it can be set to anything. There is are global setup and teardown files to run databases, docker, etc.

## TODO

- [x] add e2e tests
- [x] add lint target for e2e project

### CODE REVIEW

File: packages/node/src/generators/application/application.ts [hunk 1]
```diff
@@ -44,6 +44,8 @@ import {
 import { prompt } from 'enquirer';
 
 import * as shared from '@nrwl/workspace/src/utils/create-ts-config';
+import e2eProjectGenerator from '@nrwl/node/src/generators/e2e-project/e2e-project';
+import { project } from '@phenomnomnominal/tsquery/dist/src/project';
```
Comment (ndcunningham) [inline packages/node/src/generators/application/application.ts:48]: Is this used anywhere?

File: packages/node/src/generators/application/files/connect/src/main.ts__tmpl__ [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 
 app.use((req, res) => {
      res.end(JSON.stringify({ 'message': 'Hello API' }));
 });
 
-app.listen(<%= port %>, () => {
-    // Server is running
-})
\ No newline at end of file
+app.listen(<port, () => {
```
Comment (ndcunningham) [inline packages/node/src/generators/application/files/connect/src/main.ts__tmpl__:11]: should be `app.listen(port, () => {` you have a `<` beside port

File: packages/node/src/generators/application/application.ts (hunk 1 above)
Comment (jaysoo) [reply packages/node/src/generators/application/application.ts]: no, removed (intellij auto-imports 😓 )

File: packages/node/src/generators/e2e-project/e2e-project.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      tsConfigPaths: [joinPathFragments(options.projectRoot, 'tsconfig.json')],
+      eslintFilePatterns: [`${options.projectRoot}/**/*.{js,ts}`],
+      setParserOptionsProject: false,
+      skipPackageJson: false,
+      rootProject: options.rootProject,
+    });
+    tasks.push(linterTask);
+  }
+
+  await formatFiles(host);
```
Comment (ndcunningham) [inline packages/node/src/generators/e2e-project/e2e-project.ts:92]: should this have `if(!formatFiles)`?

File: packages/node/src/generators/e2e-project/schema.d.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export interface Schema {
+  project: string;
+  projectType: 'server' | 'cli';
+  directory?: string;
+  name?: string;
+  port?: number;
+  linter?: 'eslint' | 'none';
+  rootProject?: boolean;
```
Comment (ndcunningham) [inline packages/node/src/generators/e2e-project/schema.d.ts:8]: missing `formatFiles: boolean` here

**Review by ndcunningham — APPROVED** [review]

**Pull request merged**